source/server/http_dynamic_delay_filter_config.cc

```diff
@@ -0,0 +1,55 @@
+#include <string>
+
+#include "envoy/registry/registry.h"
+
+#include "external/envoy/source/common/protobuf/message_validator_impl.h"
+
+#include "api/server/response_options.pb.h"
+#include "api/server/response_options.pb.validate.h"
+
+#include "server/http_dynamic_delay_filter.h"
+
+namespace Nighthawk {
+namespace Server {
+namespace Configuration {
+
+class HttpDynamicDelayDecoderFilterConfig
```

Comment:
I'm confused.  Why are we defining a class with public methods in a .cc file, rather than a .h file?

Reply:
I got confused as well when looking closer at this; but more because we had two classes named `HttpDynamicDelayDecoderFilterConfig` in two namespaces. 
Renamed this one to `HttpDynamicDelayDecoderFilterConfigFactory` which is more accurate.

As for your question on the class with public methods in the `.cc` file: when done like that, it allows other code within the same `.cc` to consume its public methods. But not publishing it in the .h prevents anything else from doing so.

Reply:
Ah, okay.  Would it be okay if we moved it within an anonymous namespace?  That seems to be the google convention for indicating what you're talking about.  If there's a reason not to do that, totally understand.

Reply:
Ooh, yeah, good catch. Done.